### PR TITLE
Add KBD67 MKII Solderable PCB Variant

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the KBDfans KBD67 MKII solderable variant PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for the KBDfans KBD67 MKII solderable variant PCB.

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbd67/mkii_soldered/keymaps/ai03/keymap.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_PSCR, \
+	KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_SPC,           KC_SPC,           KC_GRV,  KC_DEL,           KC_LEFT, KC_DOWN, KC_RGHT  \
+  ),
+  [1] = LAYOUT( /* FN */
+	RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, _______, \
+	KC_CAPS, _______, KC_PGUP, _______, _______, _______, _______, _______, KC_UP,   _______, KC_MPRV, KC_MPLY, KC_MNXT,          _______, _______, \
+	_______, KC_HOME, KC_PGDN, KC_END,  _______, KC_VOLD, KC_VOLU, KC_LEFT, KC_DOWN, KC_RGHT, _______, _______,          _______,          BL_INC,  \
+	_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          KC_PGUP, BL_DEC,  \
+	_______, _______, _______,          _______,          _______,          _______,          _______, _______,          KC_HOME, KC_PGDN, KC_END   \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/keymaps/ai03/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/kbdfans/kbd67/mkii_soldered/keymaps/ai03/keymap.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/keymaps/ai03/keymap.c]: These functions can all be removed as they aren't doing anything.

File: keyboards/kbdfans/kbd67/mkii_soldered/rules.mk [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/rules.mk]: Please remove these lines.

File: keyboards/kbdfans/kbd67/mkii_soldered/rules.mk [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/rules.mk]: And these lines, too.

File: keyboards/kbdfans/kbd67/mkii_soldered/rules.mk [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/kbdfans/kbd67/mkii_soldered/rules.mk [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkii_soldered/rules.mk]: Please replace these lines with:

```make
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

(removed size estimates, and fixed comment alignment)

**Comment by ai03-2725** [comment]
Adjusted as recommended

File: keyboards/kbdfans/kbd67/mkii_soldered/config.h [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+*/
+#define MATRIX_ROW_PINS { B3, D0, D1, D2, D3 }
+#define MATRIX_COL_PINS { F0, F1, F4, F5, F6, F7, C7, C6, B6, B5, B4, D7, D6, D4, D5 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
```
Comment (yanfali) [inline keyboards/kbdfans/kbd67/mkii_soldered/config.h]: Can probably remove these two lines, it's not a split

File: keyboards/kbdfans/kbd67/mkii_soldered/info.json [hunk 8]
```diff
@@ -0,0 +1,83 @@
+{
+    "keyboard_name": "",
```
Comment (yanfali) [inline keyboards/kbdfans/kbd67/mkii_soldered/info.json]: Fill this in please

File: keyboards/kbdfans/kbd67/mkii_soldered/info.json [hunk 9]
```diff
@@ -0,0 +1,83 @@
+{
+    "keyboard_name": "", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd67/mkii_soldered/info.json]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/kbdfans/kbd67/mkii_soldered/mkii_soldered.h [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd67/mkii_soldered/mkii_soldered.h]:
```suggestion
#define LAYOUT_all( \
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**